### PR TITLE
[ASCII-1319] Retry status command in status e2e tests

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Wrap the status test in an `EventuallyWithT`.

### Motivation

The test started to flake due to the process-agent and trace-agent not being ready when we run the status command.
Adding a grace period should solve it.

### Additional Notes

IMO the framework shouldn't start the test until all agents that need to be ready are, but this is not how we currently do it in containers for example.

### CODE REVIEW

File: test/new-e2e/tests/agent-subcommands/status/status_common_test.go [hunk 1]
```diff
@@ -180,9 +181,12 @@ func (v *baseStatusSuite) TestDefaultInstallStatus() {
...
+	require.EventuallyWithT(v.T(), func(t *assert.CollectT) {
+		status := v.Env().Agent.Client.Status()
 
-	for _, section := range expectedSections {
-		verifySectionContent(v.T(), status.Content, section)
-	}
+		for _, section := range expectedSections {
+			verifySectionContent(t, status.Content, section)
+		}
+	}, 2*time.Minute, 20*time.Second)
```
Comment (louis-cqrl) [inline test/new-e2e/tests/agent-subcommands/status/status_common_test.go:191]: How do you compute the time here, it is by experience or there is someway to compute it ?

**Review by ogaca-dd — APPROVED** [review]

File: test/new-e2e/tests/agent-subcommands/status/status_common_test.go [hunk 2]
```diff
@@ -59,7 +60,7 @@ type expectedSection struct {
 }
 
 // verifySectionContent verifies that a specific status section behaves as expected (is correctly present or not, contains specific strings or not)
-func verifySectionContent(t *testing.T, statusOutput string, section expectedSection) {
+func verifySectionContent(t require.TestingT, statusOutput string, section expectedSection) {
```
[the remark this answers is not in the document]
Comment (pgimalac) [reply test/new-e2e/tests/agent-subcommands/status/status_common_test.go:63]: `*testing.T` is a `struct`, which means you can't use anything else instead...
`require.TestingT` is an interface that `*testing.T` implements, so using that allows to give different implementations of `T`.

In particular when using `EventuallyWithT` the first argument of the lambda is of type `*assert.CollectT`, which also implements `require.TestingT`
So using `require.TestingT` allows using both implementations (if you don't need any function that is specific to either)

[the remark this answers is not in the document]
Comment (louis-cqrl) [reply test/new-e2e/tests/agent-subcommands/status/status_common_test.go:63]: thxxx

File: test/new-e2e/tests/agent-subcommands/status/status_common_test.go (hunk 1 above)
Comment (pgimalac) [reply test/new-e2e/tests/agent-subcommands/status/status_common_test.go:191]: It's just a (somewhat) random value 😄 
We don't have any guarantees on how long each agent takes to boot, and checking that it not the goal of this test, so I set something super high (2 minutes) just to be safe

Comment (louis-cqrl) [reply test/new-e2e/tests/agent-subcommands/status/status_common_test.go:191]: okay thanks

**Pull request merged**